src/sinks/aws_cloudwatch_logs/config.rs

```diff
@@ -146,7 +148,8 @@ impl SinkConfig for CloudwatchLogsSinkConfig {
     }
 
     fn input(&self) -> Input {
-        Input::log()
+        let encoder_input_type = self.encoding.clone().config().input_type();
+        Input::new(DataType::Log & encoder_input_type)
```

Comment:
I'm not against including this since it does maintain compatibility, but it seems like with the new codec system, we could loosen the requirements and, for example, let metrics be encoded as JSON and sent to CloudWatch Logs?

Reply:
Hm, yes on a first glance that would work 🤔

Haven't thought too much about if there's any cases where this could break, though

Reply:
I couldn't find any argument against that, and like the flexibility of it.

Implemented with abee634aa0f9b0eb136e65b3a62ccb1aa1ee6ac8